### ISSUE TITLE
[BUG] .Net Framework application crash after upgrade to 2.80.1 from 1.68.3

### ISSUE THREAD

**Issue by JackLi-tg** [mono/SkiaSharp#1409]
**Description**
After update from 1.68.3 to 2.80.1, application randomly either crash or throw following error: 
Managed Debugging Assistant 'FatalExecutionEngineError' : "'The runtime has encountered a fatal error. The address of the error was at 0x74a40bf1, on thread 0x3bac. The error code is 0xc0000005. This error may be a bug in the CLR or in the unsafe or non-verifiable portions of user code. Common sources of this bug include user marshaling errors for COM-interop or PInvoke, which may corrupt the stack.'

Attached a sample project to reproduce the issue.

The crash/error happens with following configurations:
1. **.Net Framework** project  (framework 4.6.2 - 4.8)
2. In project property dialog (Project property - > Build tab)
- Platform Target set to: **Any CPU**  
- **AND** "Prefer 32-bit" option **CHECKED**

**Code**

Attached a simple sample project to reproduce 
[SkiaSharp Crash Bug.zip](https://github.com/mono/SkiaSharp/files/4926432/SkiaSharp.Crash.Bug.zip)

**Actual Behavior**

The app randomly either crash or throw error.

**Basic Information**

- Version with issue:  2.80.1
- Last known good version:  1.68.3
- IDE:  Visual Studio 2019

**Screenshots**
 
![Skiasharp2 80 error](https://user-images.githubusercontent.com/63121477/87568444-46df1400-c68b-11ea-858a-96119c5d88e9.png)

**Reproduction Link**
Sample project to reproduce the issue.
[SkiaSharp Crash Bug.zip](https://github.com/mono/SkiaSharp/files/4926432/SkiaSharp.Crash.Bug.zip)

**Comment by maiad** [comment]
I have the same problem !

**Comment by mattleibow** [comment]
Thanks for the issue and repro. Having a look now.

**Comment by mattleibow** [comment]
I think this is just for 32-bit. When I set the target to 64, it works. When I set it to 32, it fails. When I set Any CPU with prefer 32 bit is fails.

**Comment by JackLi-tg** [comment]
In fact set target to 64 or 32 both works,  only fails when set AnyCPU with prefer 32 bit.
please be sure to manually delete bin and obj folders when switching targets between 32 and 64.

**Comment by mattleibow** [comment]
This is very weird. Had to accurately reproduce. The first times it throws. Then it starts just to log. Then it moves to only logging when the app closes. Then just works and no error exit code at all.

Seems the more I run it, the less crashy it becomes.

**Comment by mattleibow** [comment]
The code that I was using:

```csharp
SKPaint paint = new SKPaint();
paint.Typeface = SKTypeface.CreateDefault();
// Erros only occurs when project property - > Build tab
// 1. Platform Target set to: Any CPU 
// 2. AND "Prefer 32-bit" option CHECKED
var textLength = paint.MeasureText("test");

MessageBox.Show("Success - " + textLength);
```

**Comment by mattleibow** [comment]
I think I may have found the issue, thanks to @sremiger1 in #1449. Just doing some tests now, but it might have to do with the way I am overriding the paint object in C++. Gathering more info now and will update.

**Comment by mattleibow** [comment]
😭

**Comment by mattleibow** [comment]
@JackLi-tg @sremiger1, it appears this is a .NET Framework bug. Can you try and see if you use .NET Framework 4.8?

I will make sure to find a fix for this for net462+, but in the meantime let me know if net48 works for you.

**Comment by mattleibow** [comment]
Another thing, this seems to work in _some_ versions of net48.

For example, running this code on my machine:

```csharp
public static Version GetRunningFrameworkVersion()
{
    var netVer = Environment.Version;
    Assembly assObj = typeof(Object).GetTypeInfo().Assembly;
    if (assObj != null)
    {
        AssemblyFileVersionAttribute attr;
        attr = (AssemblyFileVersionAttribute)assObj.GetCustomAttribute(typeof(AssemblyFileVersionAttribute));
        if (attr != null)
        {
            netVer = new Version(attr.Version);
        }
    }
    return netVer;
}
```

Mu value is `4.8.4180.0` and it does not work. However, I had a colleague try it and their machine was `4.8.4075.0` and it worked there.

**Comment by mattleibow** [comment]
Just a link of some code that can reproduce this every time: https://gist.github.com/mattleibow/33f54a166cd6bdcad8580b1f43e45077

**Comment by sremiger1** [comment]
Unfortunately I am writing in netstandard and using core 2.1

**Comment by mattleibow** [comment]
Can you try netcore 3.1? That works for me. Let me install netcore 2.1. What is your specific SDK version so I can test that?

**Comment by mattleibow** [comment]
I think I have found a fix - which is more a workaround in the library as this appears to be a bug in .NET - and will push that out as soon as CI finishes.

**Comment by mattleibow** [comment]
I merged the fixes, will push out soon.

**Comment by JackLi-tg** [comment]
> @JackLi-tg @sremiger1, it appears this is a .NET Framework bug. Can you try and see if you use .NET Framework 4.8?
> 
> I will make sure to find a fix for this for net462+, but in the meantime let me know if net48 works for you.

@mattleibow Yes, net48 having the same error as net462

**Comment by mattleibow** [comment]
@JackLi-tg can you try the latest 2.80.2-preview.24 on the preview feed. This should have the fix. If that works for you, then I'll push to NuGet.org.

**Comment by JackLi-tg** [comment]
> @JackLi-tg can you try the latest 2.80.2-preview.24 on the preview feed. This should have the fix. If that works for you, then I'll push to NuGet.org.

2.80.2-preview.24 is not available yet as of Aug 10th, I'll try when it is available in Nuget

**Comment by JackLi-tg** [comment]
@mattleibow Would like to follow up if there is any update on 2.80.2-preview.24, any idea when it will be available to test with?

**Comment by mattleibow** [comment]
@JackLi-tg  sorry for the delay, there were some other things that needed to go in. However, I just pushed preview 33 to NuGet.org and it is indexing as we speak!

**Comment by mattleibow** [comment]
Just so that you know, all commits on master (and some PRs) get pushed to the preview feed: `https://aka.ms/skiasharp-eap/index.json`

You should be able to use that for any testing.

**Comment by JackLi-tg** [comment]
Verified this bug been fixed in v2.80.2
Thanks!
